Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -756,8 +756,12 @@ void SVGSMILElement::addInstanceTime(BeginOrEnd beginOrEnd, SMILTime time, SMILT
     SMILTime elapsed = this->elapsed();
     if (elapsed.isUnresolved())
         return;
+    SMILTimeWithOrigin timeWithOrigin(time, origin);
+    // Ignore new instance times for 'end' if the element is not active and the origin is script.
+    if (beginOrEnd == End && m_activeState == Inactive && timeWithOrigin.originIsScript())
+        return;
     auto& list = beginOrEnd == Begin ? m_beginTimes : m_endTimes;
-    list.append(SMILTimeWithOrigin(time, origin));
+    list.append(timeWithOrigin);
```

Comment:
I do not think this is the right place to add this code. Since `SVGAnimationElement::endElementAt()` is the caller to `addInstanceTime()` which passes `End` and `SMILTimeWithOrigin::ScriptOrigin`, I think you can do the following:

In the `protected` section of `SVGSMILElement`, add this inline function:

```
ActiveState activeState() const { return m_activeState; }
```

And in SVGAnimationElement::endElementAt() add this if statement:

```
if (activeState() == Inactive)
    return;
addInstanceTime(End, elapsed() + offset, SMILTimeWithOrigin::ScriptOrigin);
```

Reply:
@shallawa - I get compile errors with suggested approach. :-( Will continue to explore.

Reply:
I think you need to move the definition of `ActiveState` from the `private:` section to the `protected:` section:

```
protected:
    enum ActiveState { Inactive, Active, Frozen };
    ActiveState activeState() const { return m_activeState; }
    void setInactive() { m_activeState = Inactive; }
```

Reply:
Done in https://github.com/WebKit/WebKit/commit/0e646703be2b2c1f7bf2a8e6f34edf514887e5a1. Thanks for all the help @shallawa .